TATB : 06 Change the JSON file with Proper Format and get Options randomly in the quiz.

SohamDevlani opened the pull request:
-Change the JSON file with Proper Format
    -Change path everywhere to get data from JSON file.
    -Get Options in the quiz randomly

Commit:
TATB-06
    -Change the JSON file with Proper Format
    -Change path everywhere to get data from JSON file.
    -Get Options in the quiz randomly

Commit:
TATB-06
    -Change the JSON file with Proper Format
    -Change path everywhere to get data from JSON file.
    -Get Options in the quiz randomly

PoojanPatelJT commented on src/app/component/dashboard/dashboard.component.ts:
```diff
@@ -9,25 +9,19 @@ import { QuestionService } from '../../service/question.service';
+  quizData = {...quizData};
   title: any;
   arrayIndex=0;
   image: any;
```
what is image is used for ?

PoojanPatelJT commented on src/app/component/dashboard/dashboard.component.ts:
```diff
@@ -9,25 +9,19 @@ import { QuestionService } from '../../service/question.service';
   title: any;
   arrayIndex=0;
   image: any;
   description: any;
```
description is used for ?

PrabhatJTDeveloper replied:
It wasnt used anywhere so removed it.

PrabhatJTDeveloper replied:
It wasnt used anywhere hence removed it

PoojanPatelJT commented on src/app/component/dashboard/dashboard.component.ts:
```diff
@@ -36,19 +30,10 @@ export class DashboardComponent implements OnInit,OnDestroy {
-    // console.log("title",this.title)
-    // console.log("image",this.image)
-    // console.log("description",this.description)
+  formatData() {
```
Could you please remove this function as there is nothing in this

PrabhatJTDeveloper replied:
yes i removed it

PoojanPatelJT commented on src/app/component/dashboard/dashboard.component.ts:
```diff
@@ -36,19 +30,10 @@ export class DashboardComponent implements OnInit,OnDestroy {
-  // }
+
   quizname(title:any){
     console.log(title);
```
Please remove console from here

PrabhatJTDeveloper replied:
yes i removed it

PoojanPatelJT commented on src/app/component/dashboard/dashboard.component.ts:
```diff
@@ -36,19 +30,10 @@ export class DashboardComponent implements OnInit,OnDestroy {
+
   quizname(title:any){
     console.log(title);
     this.questionService.selectedQuizType = title;
```
I guess service is not in use please remove this

PrabhatJTDeveloper replied:
yes i removed it

PoojanPatelJT commented on src/app/component/quiz/quiz.component.ts:
```diff
@@ -67,6 +54,7 @@ export class Quizcomponent implements OnInit, OnDestroy {
   timer: any;
   selectedQuizType: any;
   radioValue: any;
```
Could you please check and remove if not in use

PrabhatJTDeveloper replied:
Yes i removed all unnecessary declarations

PoojanPatelJT commented on src/app/component/quiz/quiz.component.ts:
```diff
@@ -20,28 +20,15 @@ import { DialogService } from 'src/app/dialog-service/dialog.service';
-  title!: any;
   dialogData = { ...dialogData };
   private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
 
```
Please do code indentation

PrabhatJTDeveloper replied:
Done

PoojanPatelJT commented on src/app/component/quiz/quiz.component.ts:
```diff
@@ -215,17 +210,18 @@ export class Quizcomponent implements OnInit, OnDestroy {
         this.inCorrectanswer++;
       }
-      if (questionindex === this.question.length) {
+      debugger;
```
Could you please remove debugger from here

PrabhatJTDeveloper replied:
yes i removed it

PoojanPatelJT commented on src/app/component/result/result.component.ts:
```diff
@@ -8,7 +8,7 @@ import quizData from 'src/assets/json/data.json';
 export class ResultComponent implements OnInit, OnDestroy {
-  quizData: any = { ...quizData };
+  quizData = { ...quizData };
   isQuizCompleted: any;
```
what is isQuizCompleleted is used for ?

PrabhatJTDeveloper replied:
it wasnt used anywhere so removed it

PoojanPatelJT commented on src/app/component/result/result.component.ts:
```diff
@@ -8,7 +8,7 @@ import quizData from 'src/assets/json/data.json';
-  quizData: any = { ...quizData };
+  quizData = { ...quizData };
   isQuizCompleted: any;
   userName: any;
```
Could you please check and remove userName if not in  use

PrabhatJTDeveloper replied:
Yes i removed it.

PoojanPatelJT commented on src/app/component/result/result.component.ts:
```diff
@@ -8,7 +8,7 @@ import quizData from 'src/assets/json/data.json';
+  quizData = { ...quizData };
   isQuizCompleted: any;
   userName: any;
   points: any;
```
Same with this

PoojanPatelJT replied:
correctAnswer: any;
  inCorrectAnswer: any; 
  
  Please remove this function

PoojanPatelJT replied:
Please remove questionservice

PrabhatJTDeveloper replied:
i removed all unnecessary functions

PoojanPatelJT commented on src/app/component/startquiz/startquiz.component.html:
```diff
@@ -4,7 +4,7 @@
     <h2 class="card-title text-center"> Welcome to Quiz App</h2>
      <h4 class="card-subtitle"> Rules:</h4>
-     <ol [innerHTML]="rules">
+     <ol [innerHTML]="rules.rules">
```
Could you please change variable name

PrabhatJTDeveloper replied:
i changed variable name.

PoojanPatelJT commented on src/app/component/startquiz/startquiz.component.ts:
```diff
@@ -24,11 +24,8 @@ export class StartquizComponent implements OnInit {
-   this.rules = this.quizData[this.selectedQuizType]?.rules;
-   console.log("this.rules ----------",this.rules)
+    this.selectedQuizType = this.activeRoute.snapshot.queryParams['quiz'];
+   this.rules = this.quizData.Quiz.find((data)=> data.quizId == this.selectedQuizType );
```
Please add null check here

PrabhatJTDeveloper replied:
i added the null check

charmijtdev committed:
TATB-06
    -Change the JSON file with Proper Format
-Change path everywhere to get data from JSON file.
-Get Options in the quiz randomly

charmijtdev committed:
TATB-06
    -Change the JSON file with Proper Format
-Change path everywhere to get data from JSON file.
-Get Options in the quiz randomly

PrabhatJTDeveloper committed:
TATB-06 : Change JSON file format
      - Give appropriate function names
      -Manage the positive and negative points from JSON file.
      -find another way of answer other then using value
      -Manage number of questions in a single quiz from JSON file.

PrabhatJTDeveloper committed:
TATB-06 : Change JSON file format
      - Give appropriate function names
      -Manage the positive and negative points from JSON file.
      -find another way of answer other then using value
      -Manage number of questions in a single quiz from JSON file.

PoojanPatelJT commented on src/app/component/quiz/quiz.component.ts:
```diff
@@ -20,53 +20,31 @@ import { DialogService } from 'src/app/dialog-service/dialog.service';
   selectedQuizType: any;
   radioValue: any;
+  options: string | number | undefined;
+  positivePoints: any;
```
correctAnswer: number = 0;
  wrongAnswer: number = 0; 
  Why you have declared this  ? As in above i can see those variable

charmijtdev replied:
This is uesd three time Please check 
correctanswer: number = 0;

PrabhatJTDeveloper replied:
yes i have removed unnecessary variables

PoojanPatelJT commented on src/app/component/quiz/quiz.component.ts:
```diff
@@ -99,11 +77,27 @@ export class Quizcomponent implements OnInit, OnDestroy {
+    this.options = this.question.map((question: any) =>
+      question.options.sort(() => Math.random() - 0.69)
+    );
+    this.timer = this.quizData.Quiz.find(
```
Can we do all this in one ?

PrabhatJTDeveloper committed:
TATB -35 :Show only recently completed result then create option for showing all results
      -[BUG] Getting 2 results of same quiz
      -Change the card design of all results page
      -[BUG] Data does not remain after refresh
      -Get data of completed quiz
      -implement initial letter of username as display picture
      -implement show more button in all results

PoojanPatelJT commented on src/app/component/dashboard/dashboard.component.ts:
```diff
@@ -1,57 +1,25 @@
-    this.QuizType = this.quizData?.QuizType;
-
-    // this.mapJSONData();
+    this.QuizType = this.quizData.Quiz;
```
please use camelCase for the varaibles

PrabhatJTDeveloper replied:
yes i used camelcase for the variables

PoojanPatelJT commented on src/app/component/quiz/quiz.component.ts:
```diff
@@ -20,53 +20,31 @@ import { DialogService } from 'src/app/dialog-service/dialog.service';
-
-  selectedOption!: boolean;
-
   questionindex = 0;
```
please give proper datatypes for questionIndex, counter, intervals$ variables

PrabhatJTDeveloper replied:
yes i gave proper data types

PoojanPatelJT commented on src/app/component/quiz/quiz.component.ts:
```diff
@@ -99,11 +77,27 @@ export class Quizcomponent implements OnInit, OnDestroy {
+    )?.NegativePoints;
 
     const formArray = this.myForm.controls['form'] as FormArray;
     this.question.forEach((item: any) => {
```
Please use camelCase for all varaibles

PrabhatJTDeveloper replied:
yes i used camel case for all variables

charmijtdev commented on src/app/component/dashboard/dashboard.component.ts:
```diff
@@ -1,57 +1,25 @@
 export class DashboardComponent implements OnInit,OnDestroy {
-  quizData = quizData;
+  quizData = {...quizData};
   title: any;
```
Please give proper format like string

PrabhatJTDeveloper replied:
yes i gave proper format to every variable

charmijtdev commented on src/app/component/dashboard/dashboard.component.ts:
```diff
@@ -1,57 +1,25 @@
-  image: any;
-  description: any;
-  QuizType:any = [0,1,2];
+  QuizType:any;
```
here too and please give name like quizType:string

PrabhatJTDeveloper replied:
yes i have given type and also changed the name

charmijtdev commented on src/app/component/dashboard/dashboard.component.ts:
```diff
@@ -1,57 +1,25 @@
-  //   this.imagesssindex++;
-  // }
+
   quizname(title:any){
```
Please use camel case name standard and please change like routeToQuiz

PrabhatJTDeveloper replied:
yes i have given camelcase name and changed name

charmijtdev commented on src/app/component/quiz/quiz.component.ts:
```diff
@@ -20,53 +20,31 @@ import { DialogService } from 'src/app/dialog-service/dialog.service';
   interval$: any;
-  validate: any;
   points: number = 0;
   correctanswer: number = 0;
```
Please check again  
correctAnswer: number = 0;
  wrongAnswer: number = 0;

PrabhatJTDeveloper replied:
yes i have removed it

charmijtdev commented on src/app/component/quiz/quiz.component.ts:
```diff
@@ -20,53 +20,31 @@ import { DialogService } from 'src/app/dialog-service/dialog.service';
-
-  selectedOption!: boolean;
-
   questionindex = 0;
```
could you please change name to questionindex to questionIndex

PrabhatJTDeveloper replied:
Yes i changed the name

charmijtdev commented on src/app/component/quiz/quiz.component.ts:
```diff
@@ -20,53 +20,31 @@ import { DialogService } from 'src/app/dialog-service/dialog.service';
   filterquestionsindex: any;
-  checkIfselctedanswer: any;
-  userName: any;
   timer: any;
```
Here please remove timer as counter is used upper and give type

PrabhatJTDeveloper replied:
i couldnt remove timer as data of timer is coming from json file but counter was able to be removed so removed it

charmijtdev commented on src/app/component/startquiz/startquiz.component.ts:
```diff
@@ -9,8 +9,8 @@ import quizData from '../../../assets/json/data.json';
-  rules: any;
+  quizData = {...quizData};
+  instruction: any;
   selectedQuizType: any;
```
Please give type string

PrabhatJTDeveloper replied:
yes i gave

charmijtdev commented on src/assets/json/data.json:
```diff
@@ -1,1107 +1,1110 @@
+      "PositivePoints": 10,
+      "NegativePoints": 1,
+      "numberOfQuestions": "3",
+      "type": "radio",
```
what is use for type here ?

PrabhatJTDeveloper replied:
type is given for future enhancement as we could change the option type from radio button to checbox.

charmijtdev commented on src/assets/json/data.json:
```diff
@@ -1,1107 +1,1110 @@
         },
         {
-          "id": "Q-11.",
+          "questionId": "Q-11.",
```
what is use of this question id ?

PrabhatJTDeveloper replied:
There was no use of questionId hence removed

PoojanPatelJT committed:
Reverting changes and making develop branch up to date  (#18)

* -revert Chagnes

* -revert changes to scss

* -added angualr file

* -changes css type

PrabhatJTDeveloper committed:
TATB - 06 Change the JSON file with Proper Format and get Options randomly in the quiz.
     -give all variables proper datatypes
     -change all variables names into camelcase names
     -remove unnecessary variables
     -make the dashboard responsive until mobile view

PrabhatJTDeveloper committed:
TATB - 06 Change the JSON file with Proper Format and get Options randomly in the quiz.
     -give all variables proper datatypes
     -change all variables names into camelcase names
     -remove unnecessary variables
     -make the dashboard responsive until mobile view

PoojanPatelJT committed:
Bug/revert changes (#19)

* -revert Chagnes

* -revert changes to scss

* -added angualr file

* -changes css type

* -changes

* changes revert for gitignore

PrabhatJTDeveloper committed:
Merge branch 'develop' of https://github.com/Jyoti-Technosoft/technical-assistant into feature/TATB-06-change-json-file-format-and-get-random-question

PrabhatJTDeveloper committed:
Merge branch 'feature/TATB-06-change-json-file-format-and-get-random-question' of https://github.com/Jyoti-Technosoft/technical-assistant into feature/TATB-06-change-json-file-format-and-get-random-question

charmijtdev committed:
-changed git ignore

charmijtdev committed:
-chagned git igonore changes

charmijtdev committed:
sqashed commits

charmijtdev committed:
Merge branch 'develop' of https://github.com/Jyoti-Technosoft/technical-assistant into develop

PoojanPatelJT committed:
Merge branch 'develop' into feature/TATB-06-change-json-file-format-and-get-random-question

charmijtdev commented on src/app/component/allresults/allresults.component.html:
```diff
@@ -0,0 +1,62 @@
+<div class="col text-center mb-3" style="margin-top: 50px">
```
Could you plese remove this inline css ?

PrabhatJTDeveloper replied:
yes i removed the inline css

charmijtdev commented on src/app/component/allresults/allresults.component.ts:
```diff
@@ -0,0 +1,68 @@
+  constructor(public questionService: QuestionService) {}
+
+  ngOnInit(): void {
+    this.userName = this.questionService?.userName;
```
No need to check here ?

PrabhatJTDeveloper replied:
okay i removed it.

charmijtdev commented on src/app/component/dashboard/dashboard.component.html:
```diff
@@ -1,94 +1,40 @@
+  <div class="row">
+    <div class="min-vh-100 d-flex align-items-center">
+      <div class="card   mt-5" id="card">
+        <div class="card-body" style="padding: 0px">
```
Please remove inline css

PrabhatJTDeveloper replied:
Yes i removed the inline css

charmijtdev commented on src/app/component/quiz/quiz.component.html:
```diff
@@ -45,12 +44,12 @@
+=======
 
-              <ng-container *ngFor="let card of question[this.questionindex].options">
+>>>>>>> c2bb962d932820cb3282424f75c3ea93d58f0eac
```
Please also remove this

PrabhatJTDeveloper replied:
Yes i removed it

charmijtdev commented on src/app/component/quiz/quiz.component.html:
```diff
@@ -45,12 +44,12 @@
-            
-              </canvas> -->
 
+<<<<<<< HEAD
```
remove thsi confict

PrabhatJTDeveloper replied:
yes i removed the conflict

charmijtdev commented on src/app/component/quiz/quiz.component.ts:
```diff
@@ -157,75 +149,69 @@ export class Quizcomponent implements OnInit, OnDestroy {
       let selectedAnswer =
-        this.question[questionindex].answer?.[0] == correctOptions;
+        this.question[questionIndex].answer.id == correctOptions;
       if (selectedAnswer && correctOptions) {
```
No need to create a comparision your can direct check this condition here

charmijtdev commented on src/app/component/result/result.component.html:
```diff
@@ -1,88 +1,49 @@
-    <div class="row g-0">
-      <div class="col text-center mb-3 mt-4">
-        <span class="fs-3">Result</span>
+<div class="col-lg-5 col-md-7 col-sm-12 col-xs-12" style="margin: 75px auto; float: none; margin-bottom: 10px;">
```
Please remove this inline css here

PrabhatJTDeveloper replied:
Yes i removed the inline css

charmijtdev commented on src/app/component/result/result.component.html:
```diff
@@ -1,88 +1,49 @@
-       </h1>
+    <div class="container">
+    <div class="row">
+      <div class="text-center" style="margin-bottom: 30px;">
```
Here also please remove this

PrabhatJTDeveloper replied:
Yes i removed the inline css

PrabhatJTDeveloper committed:
-Removed inline css
-removed unnecessary variables

PoojanPatelJT commented on src/app/component/allresults/allresults.component.css:
```diff
@@ -0,0 +1,288 @@
+}
+
+.all-result .quiz-points .points span {
+  margin-left: .5rem;
```
is it required to use span do we use class here ?

PoojanPatelJT commented on src/app/component/allresults/allresults.component.css:
```diff
@@ -0,0 +1,288 @@
+
+.all-result .quiz-action a {
+  text-transform: uppercase;
+}
```
Same here

PoojanPatelJT commented on src/app/component/dashboard/dashboard.component.html:
```diff
@@ -1,94 +1,40 @@
+  <div class="row">
+    <div class="min-vh-100 d-flex align-items-center">
+      <div class="card   mt-5" id="card">
+        <div class="card-body">
```
remove extra space from here

PoojanPatelJT commented on src/app/component/allresults/allresults.component.ts:
```diff
@@ -0,0 +1,67 @@
+    let data: any = localStorage.getItem('registeruser');
+    this.userData = JSON.parse(data);
+    this.userName = this.userData.find((data:any)=>{
+      return data.id == this.questionService.getUser()
```
apply prettier to the file for proper indentation

PoojanPatelJT commented on src/app/component/dashboard/dashboard.component.scss:
```diff
@@ -39,9 +37,7 @@ h5{
-    height: 350px;
-}
+
 /* .card-wrapper{
```
please remove commented code from here
